src/rust/aggregator/src/pipelines/candlesticks.rs

```diff
@@ -76,10 +76,7 @@ impl Pipeline for Candlesticks {
         .await
         .map_err(|e| PipelineError::ProcessingError(anyhow!(e)))?;
 
-        transaction
-            .commit()
-            .await
-            .map_err(|e| PipelineError::ProcessingError(anyhow!(e)))?;
+        commit_transaction(transaction).await?;
```

Comment:
If you're going to update this, how about also abstracting in the new util function to create repeatable read txn?